### PR TITLE
#163672728 Create Users table model

### PR DESCRIPTION
**Pull request by Ndifreke**
**What does this PR do ?**
- Creates a User database table 

**Description of Task to be completed ?**
-  Create sequelize Users model table

**What are the relevant pivotal tracker story ?**

[#163672728](https://www.pivotaltracker.com/n/projects/2240406/stories/163672728)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,9 +13,9 @@
     "pretest": "npm run build",
     "test": "cross-env NODE_ENV=test nyc mocha dist/tests/**/index.spec.js --timeout 80000 --exit",
     "clean": "rm -rf dist",
-    "coverage": "nyc report --reporter=text-lcov|coveralls",
+    "coverage": "npm test && nyc report --reporter=text-lcov | coveralls",
```
Comment (obasajujoshua31) [inline package.json]: please can you change the coverage script to the above

File: server/models/users.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  const User = sequelize.define("Users", userSchema, flags);
+
+  User.hook("beforeCreate", instance => {
+    if (instance.dataValues.id !== null) {
+      throw new Error("User Id cannot be set manually");
+    }
+  });
+
+  User.associate = db => {
+    console.log("Association defination");
```
Comment (benfluleck) [inline server/models/users.js]: Remove this console.log statement

File: routes/api/users.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+const router = require("express").Router();
+const passport = require("passport");
+
+router.get("/user", (req, res, next) => {
+  User.findById(req.payload.id)
```
Comment (obasajujoshua31) [inline routes/api/users.js]: Please delete this routes folder outside the server, we do not need it anylonger

File: server/tests/model.specs/userModel.specs.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    let dropped = false;
+    await Users.findOne({ where: { email: "user@email.com" } }).catch(() => {
+      dropped = true;
+    });
+    assert.isTrue(dropped);
+  });
+
+  it("should reject setting user id on creation", async () => {
+    const badUser = Object.assign(userTemplate, { id: 1 });
+    expect(Users.create(badUser)).to.be.rejectedWith(Error);
```
Comment (obasajujoshua31) [inline server/tests/model.specs/userModel.specs.js:47]: your test is not consistent, either expect or assert. pls

File: package.json (hunk 1 above)
Comment (benfluleck) [reply package.json]: @obasajujoshua31  what is the above?

File: server/migrations/user.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+module.exports = {
```
Comment (benfluleck) [inline server/migrations/user.js]: use export default

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: server/tests/model.specs/userModel.specs.js (hunk 4 above)
Comment (mrkingz) [reply server/tests/model.specs/userModel.specs.js:47]: We agreed on using expect as a team

File: routes/api/users.js (hunk 3 above)
Comment (mrkingz) [reply routes/api/users.js]: Yes!

File: package.json (hunk 1 above)
Comment (Ndifreke) [reply package.json]: > @obasajujoshua31 what is the above?

Done in latest commit

File: server/tests/model.specs/userModel.specs.js (hunk 4 above)
Comment (Ndifreke) [reply server/tests/model.specs/userModel.specs.js:47]: expect BDD is used by chai-as-promise to catch rejected Promise and assert is used for TDD

File: .gitignore [hunk 6]
```diff
@@ -34,7 +34,9 @@ dist
 
 # Optional npm cache directory
 .npm
-
+dist
+.sqlite.db
+authors_haven_test.sqlite
```
Comment (benfluleck) [inline .gitignore]: You can remove this as I have instructed gitignore to ignore all sql files

**Review by eokwukwe — APPROVED** [review]

File: server/models/index.js [hunk 7]
```diff
@@ -28,7 +27,10 @@ Object.keys(db).forEach(modelName => {
   }
 });
 
+if (env === "production" || env === "PRODUCTION") {
```
Comment (benfluleck) [inline server/models/index.js:30]: What does this do and is one for production and the other SUPER PRODUCTION?  Also why for only production what about test and development?

File: server/models/users.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
+const getUserModel = (sequelize, Sequelize) => {
+  const flags = {
+    timestamps: false,
```
Comment (benfluleck) [inline server/models/users.js:3]: why timestamps false, i thought you wanted an updatedAt column?

**Review by benfluleck — CHANGES_REQUESTED** [review]

**Comment by benfluleck** [comment]
Please implement the migration file as well so the controllers can start to utilise it

**Pull request closed without merging**